### PR TITLE
Implement "Remove from Playlist" from PlaylistActionBubbleView

### PR DESCRIPTION
**Pull request by sangwoo108**
* Make new observer APIs to set parents playlist properly
* Fix bug where download item could be dangled

Resolves https://github.com/brave/brave-browser/issues/31584

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by fallaciousreasoning — APPROVED** [review]
LGTM!

**Comment by sangwoo108** [comment]
@deeppandya , FYI, I added new observer callbacks.

**Comment by deeppandya** [comment]
> @deeppandya , FYI, I added new observer callbacks.

Yeah I briefly looked at the changes. I will submit my review tonight.

File: browser/playlist/playlist_tab_helper.cc [hunk 1]
```diff
@@ -123,6 +129,41 @@ void PlaylistTabHelper::OnItemCreated(mojom::PlaylistItemPtr item) {
...
+  }
+
+  (*iter)->parents.push_back(playlist_id);
+
+  for (auto& observer : observers_) {
+    observer.OnSavedItemsChanged(saved_items_);
+  }
+}
+
+void PlaylistTabHelper::OnItemRemovedFromList(const std::string& playlist_id,
```
Comment (simonhong) [inline browser/playlist/playlist_tab_helper.cc:149]: Q) don't we need to remove item from `saved_items_` when in this callback?

Comment (sangwoo108) [reply browser/playlist/playlist_tab_helper.cc:149]: As a item could belong to multiple playlists, we shouldn't erase the item. Instead, we'll do that when OnItemDeleted() is called. When there's no playlist references a item, this callback will be called.

**Review by simonhong — APPROVED** [review]

**Review by deeppandya — APPROVED** [review]
LGTM. Thank you @sangwoo108

**Pull request merged**